enhancements/sdk-kubernetes-1.20.md

```diff
@@ -0,0 +1,248 @@
+[external] Require Helm release that supports 1.20
+[external] Require kubebuilder release that scaffolds projects using 1.20
+[internal] Require operator-framework/api release that supports 1.20
+
+
+### Goals
+
+- Operator SDK’s dependencies are upgraded to use Kubernetes 1.20.
+- All e2e tests run during Operator SDK CI are upgraded to test against Kubernetes 1.20.
+- New projects scaffolded with Operator SDK are upgraded to use Kubernetes 1.20 dependencies.
+- New projects scaffolded with Operator SDK should generate v1 CRDs by default.
+
+
+### Non-Goals
+
+- Feature development and bug fixes that are not directly related to Kubernetes 1.20 support. The
+  rubric here is, if after bumping `go.mod` dependency versions:
+  - Will the test suite fail CI without this change?
+  - Will ignoring this change cause a regression?
+
+  If the answer to these questions is no, then it is out of scope for this enhancement.
+  - Unrelated bug fixes can be done before or after this enhancement is implemented.
+  - Integration of new features introduced in 1.20 will be done in follow-on PRs.
+
+- SDK Go version bump to 1.15
+  - Risk: users are often slow to upgrade go versions, but upstream deps might use go 1.15 features.
+  - Mitigation: we will only bump go version to 1.15 if there are breaking changes (go 1.15 features) introduced by Kubernetes 1.20.
+
+## Proposal
+
+### User Stories
+
+#### Upgrade controller-runtime to support Kubernetes 1.20
+
+_GitHub: [kubernetes-sigs/controller-runtime][gh-controller-runtime]_
+
+This story consists of:
+
+- Bumping the Kubernetes dependency versions in go.mod from 1.19 to 1.20.
+- Upgrading any relevant documentation to note support for and any breaking changes introduced by
+  the bump to Kubernetes 1.20.
+- Making any necessary changes to get CI to pass.
+
+#### Upgrade controller-tools to support Kubernetes 1.20
+
+_GitHub: [kubernetes-sigs/controller-tools][gh-controller-tools]_
+
+This story consists of:
+
+- Bumping the Kubernetes dependency versions in go.mod from 1.18 to 1.20.
+- Changing the default generated CRD version in controller-gen from v1beta1 to v1.
+- Upgrading any relevant documentation to note support for and any breaking changes introduced by
+  the bump to Kubernetes 1.20.
+- Making any necessary changes to get CI to pass.
+
+#### Upgrade kubebuilder to support Kubernetes 1.20
+
+_GitHub: [kubernetes-sigs/kubebuilder][gh-kubebuilder]_
+
+This story consists of:
+
+- Bumping the controller-runtime dependency version in go.mod file scaffolds to the version of
+  controller-runtime that supports Kubernetes 1.20.
+- Bumping the controller-gen dependency version in the Makefile scaffolds to the version of
+  controller-gen that supports Kubernete 1.20.
+- Upgrading any relevant documentation to note support for and any breaking changes introduced by
+  the bump to Kubernetes 1.20.
+- Making any necessary changes to get CI to pass.
+
```

Comment:
Will be also required perform the changes/bump into https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/ which is used in the addon projects. otherwise we cannot upgrade the kubebuilder.